### PR TITLE
Return empty list of channels when requested more channels at the end of cursor

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
In the scenario of getting channels with cursor at the last channel, no error should be thrown. Instead it should return empty array of channels to indicate no more channels are left for pagination.

Discussion https://community.mattermost.com/core/pl/ijqe8oanc7nt7k6hyis8hfk89c

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Lovely, thanks for adding a unit test as well :)

**Comment by M-ZubairAhmed** [comment]
@agnivade because the function does not return not found error anymore is this still valid
https://github.com/mattermost/mattermost-server/blob/e6b06a6dae004e8a15b82a1add09ac6b1f88409d/app/channel.go#L1881-L1883

File: api4/resolver_channel_test.go [hunk 1]
```diff
@@ -310,7 +310,7 @@ func TestGraphQLChannels(t *testing.T) {
 
 		resp, err = th.MakeGraphQLRequest(&input)
 		require.NoError(t, err)
-		require.Len(t, resp.Errors, 1) // no channels found
+		require.Len(t, resp.Errors, 0) // no errors for no channels found
```
Comment (M-ZubairAhmed) [inline api4/resolver_channel_test.go:313]: because no channels doesn't not throw error anymore

**Review by noxer — APPROVED** [review]
LGTM, thanks.

**Comment by M-ZubairAhmed** [comment]
Report E2E : https://automation-dashboard.vercel.app/cycles/528cb8c4-0b16-4162-acd8-ae74f04fb416

**Comment by M-ZubairAhmed** [comment]
Diff from master e2e

- "MM-T1477 Accessibility Support in Status Dropdown"
- "MM-T1755 - Restrict Domains - Email invite"
- "MM-T644 Integrations display on team where they were created"
- "MM-T248_2 Mark DM post as Unread in a reply thread"
- "MM-T112 Delete a parent message that has a reply - reply thread"
- "MM-T209 Input box on reply thread can expand"

**Comment by M-ZubairAhmed** [comment]
I cross checked all the failing tests and verified they were due to missing elements not responsible for changes in the pr. Mostly they were flaky test.

**Comment by agnivade** [comment]
Feel free to merge at your convenience Zubair. :+1:

**Pull request merged**